api/v1beta1/cloudstackcluster_webhook_test.go

```diff
@@ -14,198 +14,85 @@ See the License for the specific language governing permissions and
-				Expect(k8sClient.Update(ctx, cloudStackClusterUpdate).Error()).Should(MatchRegexp(forbiddenRegex, "controlplaneendpointhost"))
+		It("Should reject a CloudStackCluster with IdentityRef not of kind 'Secret'", func() {
+			dummies.CSCluster.Spec.IdentityRef.Kind = "ConfigMap"
+			Ω(k8sClient.Create(ctx, dummies.CSCluster)).
+				Should(MatchError(MatchRegexp(forbiddenRegex, "must be a Secret")))
+		})
+	})
 
-				cloudStackCluster.DeepCopyInto(cloudStackClusterUpdate)
-				cloudStackClusterUpdate.Spec.IdentityRef.Kind = "ConfigMap"
-				Expect(k8sClient.Update(ctx, cloudStackClusterUpdate).Error()).Should(MatchRegexp(forbiddenRegex, "identityRef\\.Kind"))
+	Context("When updating a CloudStackCluster", func() {
+		BeforeEach(func() {
+			Ω(k8sClient.Create(ctx, dummies.CSCluster)).Should(Succeed())
+		})
 
-				cloudStackCluster.DeepCopyInto(cloudStackClusterUpdate)
-				cloudStackClusterUpdate.Spec.IdentityRef.Name = configMapName
-				Expect(k8sClient.Update(ctx, cloudStackClusterUpdate).Error()).Should(MatchRegexp(forbiddenRegex, "identityRef\\.Name"))
-			})
+		It("Should reject updates to CloudStackCluster Zones", func() {
+			dummies.CSCluster.Spec.Zones = []infrav1.Zone{dummies.Zone1}
+			Ω(k8sClient.Update(ctx, dummies.CSCluster)).Should(MatchError(MatchRegexp(forbiddenRegex, "Zones and sub")))
+		})
+		It("Should reject updates to CloudStackCluster Zones", func() {
+			dummies.CSCluster.Spec.Zones[0].Network.Name = "ArbitraryUpdateNetworkName"
+			Ω(k8sClient.Update(ctx, dummies.CSCluster)).Should(MatchError(MatchRegexp(forbiddenRegex, "Zones and sub")))
+		})
+		It("Should reject updates to CloudStackCluster controlplaneendpoint.host", func() {
+			dummies.CSCluster.Spec.ControlPlaneEndpoint.Host = "1.1.1.1"
+			Ω(k8sClient.Update(ctx, dummies.CSCluster)).
+				Should(MatchError(MatchRegexp(forbiddenRegex, "controlplaneendpoint\\.host")))
+		})
 
-			It("Should reject changing the port", func() {
-				cloudStackCluster.DeepCopyInto(cloudStackClusterUpdate)
-				cloudStackClusterUpdate.Spec.ControlPlaneEndpoint.Port = int32(1234)
-				Expect(k8sClient.Update(ctx, cloudStackClusterUpdate)).ShouldNot(Succeed()) //(forbiddenRegex, "controlplaneendpointport"))
-			})
+		It("Should reject updates to CloudStackCluster controlplaneendpoint.port", func() {
+			dummies.CSCluster.Spec.ControlPlaneEndpoint.Port = int32(1234)
+			Ω(k8sClient.Update(ctx, dummies.CSCluster)).
+				Should(MatchError(MatchRegexp(forbiddenRegex, "controlplaneendpoint\\.port")))
+		})
+		It("Should reject updates to the CloudStackCluster identity reference kind", func() {
+			dummies.CSCluster.Spec.IdentityRef.Kind = "ConfigMap"
+			Ω(k8sClient.Update(ctx, dummies.CSCluster)).
+				Should(MatchError(MatchRegexp(forbiddenRegex, "identityref\\.kind")))
+		})
+		It("Should reject updates to the CloudStackCluster identity reference name", func() {
+			dummies.CSCluster.Spec.IdentityRef.Name = "ConfigMap"
```

Comment:
Setting the Name to "ConfigMap" feels wrong, given that it's a type name, even though it works.  I'd rather the name be "NewName" or something like that.

Reply:
I just did what the previous test was doing, but I can change it.

Basically it should always be "Secret" for us.

Do you want me to change it to "NewName"?

Reply:
Yes.  Otherwise it looks like a mistake to my eyes until I study it more carefully.  I'd rather other readers don't get distracted by it.